client/service/src/task_manager/mod.rs

```diff
@@ -127,7 +124,17 @@ impl SpawnTaskHandle {
 			}
 		};
 
-		let join_handle = self.executor.spawn(Box::pin(future.in_current_span()), task_type);
+		let future: Pin<Box<dyn Future<Output = ()> + Send>> = if let Some(telemetry_span) = self.telemetry_span.clone() {
+			// this will preserve the current spans entered & enter the telemetry span
```

Comment:
Can you please explain how this preserves the current span? 

I read the documentation and that doesn't indicate this?

Reply:
Yes me neither but I tested and it does preserve all the entered spans. I suspect "a" span in the sense of tracing is all the span you entered.

Reply:
OK I will add more doc.

I investigated and I believe the reason why it works is because when you create a new span, the previous one becomes "parent" of the new one. The prefix-log span is created synchronously before anything starts, then the telemetry span is also created synchronously when `new_full_parts` is ran. Therefore the telemetry span becomes the current span and the prefix-log is its parent.

So yeah, it preserve all the entered spans because it preserve the current span which is the descendant of all the previous span that has been created.